letsencrypt/client/display/ops.py

```diff
@@ -6,41 +7,103 @@
 from letsencrypt.client import interfaces
 from letsencrypt.client.display import util as display_util
 
+
 # Define a helper function to avoid verbose code
 util = zope.component.getUtility  # pylint: disable=invalid-name
 
 
-def choose_authenticator(auths, errs):
-    """Allow the user to choose their authenticator.
+def choose_plugin(prepared, question):
+    """Allow the user to choose ther plugin.
 
-    :param list auths: Where each of type
-        :class:`letsencrypt.client.interfaces.IAuthenticator` object
-    :param dict errs: Mapping IAuthenticator objects to error messages
+    :param list prepared: List of `~.PluginEntryPoint`.
+    :param str question: Question to be presented to the user.
 
-    :returns: Authenticator selected
-    :rtype: :class:`letsencrypt.client.interfaces.IAuthenticator` or `None`
+    :returns: Plugin entry point chosen by the user.
+    :rtype: `~.PluginEntryPoint`
 
     """
-    descs = [auth.description if auth not in errs
-             else "%s (Misconfigured)" % auth.description
-             for auth in auths]
+    opts = [plugin_ep.name_with_description
```

Comment:
(As you somewhat noted) plugin names must be unique while descriptions are not. While core plugins will probably never overlap descriptions (though we currently have two DNS implementations in the queue!), this will become a problem when people write their own plugins, not included in the core.

Moreover, there is a strong UX argument that relies on the fact that unique plugin names associate different components of the system... Plugin names include the prefix (i.e. the source package). No prefix = core plugin (people are likely to put more trust in core plugins). CLI configuration args and config files include prefix. `letsencrypt plugins` displays all plugins by their unique name. `--authenticator/--installer`, that are exact counterparts of that menu we discuss, require unique name. All these features allow users to easily identify corresponding functions of the system. IMHO if we remove one of those pieces, we will become less transparent and thus more confusing.